test/transactional-mutations-plugin/transactional-mutations-listener.test.ts

```diff
@@ -0,0 +1,89 @@
+    describe('willSendResponse tests', () => {
+        it('requestContext contain errors and there is transaction active, the transaction rolledBack', () => {
+            const errors = [{ message: 'error 1' }, { message: 'error 2' }];
+            const requestContext = setUpContext(errors, undefined);
+            queryRunnerMock = setUpQueryRunnerMock(true);
+            transactionalMutationsListener = new TransactionalMutationsListener(loggerMock as any, queryRunnerMock as any);
+
+            transactionalMutationsListener.willSendResponse(requestContext);
+
+            expect(queryRunnerMock.rollbackTransaction).toHaveBeenCalledTimes(1);
+            expect(loggerMock.debug).toHaveBeenCalledTimes(2);
+            expect(loggerMock.debug).toHaveBeenCalledWith(transactionalMutationsMessages.listenerRollingBackMessage, requestContext.context);
+            expect(loggerMock.debug).toHaveBeenCalledWith(transactionalMutationsMessages.listenerFinishedJob, requestContext.context);
+        });
+
+        it('requestContext contain errors and there isn\'t transaction active, nothing happened', () => {
+            const errors = [{ message: 'error 1' }, { message: 'error 2' }];
+            const requestContext = setUpContext(errors, undefined);
+            queryRunnerMock = setUpQueryRunnerMock(false);
+            transactionalMutationsListener = new TransactionalMutationsListener(loggerMock as any, queryRunnerMock as any);
+
+            transactionalMutationsListener.willSendResponse(requestContext);
+
+            expect(queryRunnerMock.rollbackTransaction).toHaveBeenCalledTimes(0);
+            expect(loggerMock.debug).toHaveBeenCalledTimes(1);
+            expect(loggerMock.debug).toHaveBeenCalledWith(transactionalMutationsMessages.listenerFinishedJob, requestContext.context);
+        });
+
+        it('requestContext response contain errors and there is transaction active, the transaction rolledBack', () => {
+            const response = {
+                errors: [{ message: 'error 1' }, { message: 'error 2' }],
+            };
+            const requestContext = setUpContext(undefined, response);
+            queryRunnerMock = setUpQueryRunnerMock(true);
+            transactionalMutationsListener = new TransactionalMutationsListener(loggerMock as any, queryRunnerMock as any);
+
+            transactionalMutationsListener.willSendResponse(requestContext);
+
+            expect(queryRunnerMock.rollbackTransaction).toHaveBeenCalledTimes(1);
+            expect(loggerMock.debug).toHaveBeenCalledTimes(2);
+            expect(loggerMock.debug).toHaveBeenCalledWith(transactionalMutationsMessages.listenerRollingBackMessage, requestContext.context);
+            expect(loggerMock.debug).toHaveBeenCalledWith(transactionalMutationsMessages.listenerFinishedJob, requestContext.context);
+        });
+
+        it('requestContext response contain errors and there isn\'t transaction active, nothing happened', () => {
```

Comment:
You did this test twice

Reply:
once it was the requestContext contains errors, and once the requestContext response contains errors - is it the same case ? i'm not sure